### PR TITLE
Basic building blocks for Python API with fake data example, docs, and CI

### PR DESCRIPTION
**Pull request by trippersham**
This is to merge the various branches of initial Python work into staging

### CODE REVIEW

File: .github/workflows/python-ci.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Lint
+      run: |
+        source $CONDA/etc/profile.d/conda.sh
+        conda env create
+        conda activate volunteer-portal
+        mypy ./api
```
Comment (Swimburger) [inline .github/workflows/python-ci.yml:30]: I think it should be `mypy .` here. I must have missed this.

Comment (trippersham) [reply .github/workflows/python-ci.yml]: Yup, fixed it!

**Review by Swimburger — APPROVED** [review]
Looks good.

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -8,6 +8,10 @@ on:
   pull_request:
```
Comment (jtuckerk) [inline .github/workflows/CI.yml:8]: Let's go ahead and change this name to node-CI in this PR?

**Review by jtuckerk — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -20,26 +20,28 @@ To run this code locally, it must first be cloned from GitHub. This can be done
...
 
-### `npm install` or `npm i`
+In the _client_ directory, you can run:
+
+#### `npm install` or `npm i`
 
 Installs the packages required to run this project locally. This is the first command developers must run after cloning.
 
-### `npm start`
+#### `npm start`
```
Comment (ecaoile) [inline README.md:31]: We'll need to change this since npm start now does something else. I can address this in a separate PR, though.

**Review by ecaoile — APPROVED** [review]

File: api/models.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    HIGH = 'high'
+    MEDIUM = 'medium'
+    LOW = 'low'
+
+class VolunteerRole(BaseModel):
+    position_id: str
+    hero_image_url: Url
+    priority: Priority
+    signup_url: Url
+    hrff_team_lead: Optional[Person]
```
Comment (MiniguyBrendan) [inline api/models.py:22]: Not that it matters *that* much, but is this meant to be `hfrr_team_lead` as opposed to `hrff_team_lead`?

Also, would it be helpful to set up some best practices regarding when we want to make alias types (i.e. `Url` and `MarkdownText`) or are we satisfied with a "we'll know it when we see it" approach?

**Review by MiniguyBrendan — APPROVED** [review]

**Pull request merged**